### PR TITLE
Windows File Handle Leak

### PR DESCRIPTION
**Pull request by stsantilena**
This updates the OS Watcher so that all the proper events are sent from the OS. It also specifies a number of tests as being for unix only. These tests are very event specific. 

LOG-14882

### CODE REVIEW

File: common/fs/src/cache/mod.rs [hunk 1]
```diff
@@ -1833,6 +1834,7 @@ mod tests {
 
     // Deletes a file
     #[tokio::test]
+    #[cfg(unix)]
     async fn filesystem_delete_file() -> io::Result<()> {
```
Comment (dkhokhlov) [inline common/fs/src/cache/mod.rs:1838]: this is basic test to verify on windows for this fix. shall pass.

File: common/notify_stream/src/lib.rs [hunk 2]
```diff
@@ -247,6 +247,7 @@ mod tests {
     }
 
     #[tokio::test]
+    #[cfg(unix)]
     async fn test_unwatch_if_exists() {
```
Comment (dkhokhlov) [inline common/notify_stream/src/lib.rs:251]: this is basic test to verify on windows for this fix. shall pass.

**Review by dkhokhlov — CHANGES_REQUESTED** [review]

File: common/notify_stream/src/lib.rs [hunk 3]
```diff
@@ -317,6 +318,7 @@ mod tests {
     }
 
     #[tokio::test]
+    #[cfg(unix)]
     async fn test_create_write_delete() -> io::Result<()> {
```
[the remark this answers is not in the document]
Comment (stsantilena) [reply common/notify_stream/src/lib.rs:326]: Added additional, Windows specific test for this.

**Comment by stsantilena** [comment]
The fixes for this were included in #453. Closing this out.

**Pull request closed without merging**